project/SUNGOD3/README.rst

```diff
@@ -0,0 +1,89 @@
+Advanced Vector Search for RAG Systems
+======================================
+
+Basic Information
+-----------------
+This project, tentatively named "AdvancedVectorSearch", aims to develop and implement novel vector search methods for RAG systems.
+
+Vector search is a crucial component in modern information retrieval systems, particularly in RAG applications. While FAISS is a well-established library for efficient similarity search, there's room for innovation in search methodologies. 
+
+Note that the goal of this project will focus on vector search, similarity comparison function, vector data storage, and does not include matrix calculations of LLM.
+
+Problem to Solve
+----------------
+The primary challenges this project aims to address are:
+
+1. Improving search speed beyond basic linear search, especially for high-dimensional vector spaces common in RAG systems.
+2. Developing new similarity metrics that may be more suitable for specific RAG applications.
+3. Creating a flexible, extensible framework that allows for easy experimentation with different search algorithms and metrics.
+4. Balancing speed, accuracy, and memory usage in vector search operations.
+
+Prospective Users
+-----------------
+This project will benefit:
+
+1. Researchers and developers working on RAG systems.
+2. Data scientists and machine learning engineers dealing with large-scale similarity search problems.
+3. Information retrieval specialists looking for customizable vector search solutions.
+4. Anyone working with high-dimensional data requiring efficient similarity search capabilities.
+5. Graduate students who lack graduation thesis.
```

Comment:
nit: who lack "a" graduation thesis.

Reply:
Your advisor should be happy if you plan to make a couple of papers, which will be a good preparation for a phd program.